content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,331 @@
+   bufferedWriter = new BufferedWriter(outputStreamWriter);  
+  while (true) { 
+
+  String msgFromClient = bufferedReader.readLine();  
+  
+  System.out.println("Client: " + msgFromClient); 
+
+  bufferedWriter.write(" MSG Received");  
+  bufferedWriter.newLine();  
+  bufferedWriter.flush();  
+  
+ if (msgFromClient.equalsIgnoreCase("BYE"))  
+                break;  
+  
+  }
+      socket.close();  
+      inputStreamReader.close();  
+      outputStreamWriter.close();  
+      bufferedReader.close();  
+      bufferedWriter.close();  
+  
+    } catch (IOException e) {  
+        e.printStackTrace();  
+  }  
+}
+```
+   
+### Methods used in ServerSocket
+
+-` public InputStream( getInputStream())` - returns an InputStream object that receives data.
+- `public OutputStream( getOutputStream())`- returns an OutputStream for sending data.
+- `public Socket accept ( )` -Waits for client connection( the program won't move to the next level until the client is connected). When you connect, you'll get a socket object that you may use to communicate with the client. 
+-  void close ( ) - closes the server socket i.e stops waiting for requests from clients.
+
+### THIS IS A SERVER PROGRAM THAT WILL BE ACCEPTING THE COMMUNICATION
+```java
+ 
+package com. company; 
+ 
+import java.io.*;  
+import java.net.ServerSocket;  
+import java.net.Socket;  
+  
+public class Server {  
+  
+    public static void main(String[] args) throws IOException {   
+  Socket socket ;  
+  InputStreamReader inputStreamReader ;  
+  OutputStreamWriter outputStreamWriter ;  
+  BufferedReader bufferedReader ;  
+  BufferedWriter bufferedWriter ;  
+  ServerSocket serversocket ;  
+  
+serversocket = new ServerSocket(5000);  
+  
+ while (true) {  
+   try {  
+         
+         socket = serversocket.accept();  
+  
+inputStreamReader = new InputStreamReader(socket.getInputStream());  
+outputStreamWriter = new OutputStreamWriter(socket.getOutputStream());  
+
+bufferedReader = new BufferedReader(inputStreamReader);  
+bufferedWriter = new BufferedWriter(outputStreamWriter);  
+
+while (true)
+{  
+
+String msgFromClient = bufferedReader.readLine();  
+
+System.out.println("Client: " + msgFromClient);   
+
+bufferedWriter.write(" MSG Received"); 
+bufferedWriter.newLine();  
+bufferedWriter.flush();  
+
+if (msgFromClient.equalsIgnoreCase("BYE"))  
+  break;  
+  }  
+
+        socket.close();  
+        inputStreamReader.close();  
+        outputStreamWriter.close();  
+        bufferedReader.close();  
+        bufferedWriter.close();  
+  
+          } catch (IOException e) {  
+            e.printStackTrace();  
+          }  
+        }  
+    }  
+}
+```
+
+The output of this  program is:
+
+```
+Connected
+Client: Hey there 
+Client: am doing this for the first time
+Client: BYE
+Closing connection
+```
+
+### Client side Programming
+# Establish a Socket Connection
+This is an instance of a socket class and can be obtained in two ways: 
```

Comment:
> This is an instance of a socket class and can be obtained in two ways:

This sounds awkward to me because I'm not sure what "this" is referring to in the sentence. I would consider rewording it to "A socket instance can be obtained in two ways:"